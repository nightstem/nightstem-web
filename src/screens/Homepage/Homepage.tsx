import Hero from '@/screens/Homepage/sections/Hero';

const Homepage = () => {
  return (
    <main className="">
      <h1 className="sr-only">Nightstem</h1>

      <Hero />
    </main>
  );
};

Homepage.displayName = 'Homepage';
export default Homepage;
