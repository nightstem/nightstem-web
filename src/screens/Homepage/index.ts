export { default as Homepage } from '@/screens/Homepage/Homepage';
