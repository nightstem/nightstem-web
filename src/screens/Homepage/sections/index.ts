export * from '@/screens/Homepage/sections/Hero';
export * from '@/screens/Homepage/sections/Philosophy';
