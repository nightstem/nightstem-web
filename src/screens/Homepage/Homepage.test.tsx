import { axe } from 'vitest-axe';
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';

import Homepage from '@/screens/Homepage/Homepage';

describe(Homepage, () => {
  it('matches the screenshot', () => {
    const { container } = render(<Homepage />);

    expect(container).toMatchSnapshot();
  });

  it('does not have accessibility violations', async () => {
    const { container } = render(<Homepage />);
    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
