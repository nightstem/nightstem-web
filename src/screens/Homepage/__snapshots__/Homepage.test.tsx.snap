// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > matches the screenshot 1`] = `
<div>
  <main
    class=""
  >
    <h1
      class="sr-only"
    >
      Nightstem
    </h1>
    <section
      class="relative grid screen-section items-center overflow-hidden px-2 lg:pl-24"
    >
      <div
        class="absolute -top-3/12 -left-1/2 -z-10 md:top-0 md:-left-1/5 aspect-square max-square-size rounded-full bg-radial from-primary-500/15 from-0% to-primary-500/0 to-100% blur-2xl"
      />
      <div
        class="screen-container"
      >
        <div
          class="flex measure-body flex-col gap-7"
        >
          <header
            class="flex flex-col gap-8"
          >
            <h2
              class="heading-xl text-primary-500 md:heading-xxl"
            >
              Quiet software grows at night.
            </h2>
            <p>
              A studio for focused software, built after hours.
            </p>
          </header>
          <div
            class="flex gap-4"
          >
            <button
              class="font-medium whitespace-nowrap select-none cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none ring-offset-dark focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:outline-none text-base px-4 py-2 rounded-md text-dark bg-primary-500 hover:bg-primary-400 active:bg-primary-400 focus-visible:ring-primary-500 disabled:opacity-60 disabled:hover:bg-primary-500 disabled:active:bg-primary-500"
              type="button"
            >
              Learn more
            </button>
            <button
              class="font-medium whitespace-nowrap select-none cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none ring-offset-dark focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:outline-none text-base px-4 py-2 rounded-md text-white bg-transparent border border-secondary-500 hover:border-secondary-400 active:border-secondary-400 hover:bg-secondary-500/20 active:bg-secondary-500/20 focus-visible:ring-secondary-500 disabled:opacity-60 disabled:hover:border-secondary-500 disabled:active:border-secondary-500"
              type="button"
            >
              View work
            </button>
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
